flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/ObserverContext.java

```diff
@@ -15,9 +15,12 @@
  * limitations under the License.
  */
 
-package org.apache.flink.kubernetes.operator.observer.context;
+package org.apache.flink.kubernetes.operator.observer;
 
-/** An empty observer context. */
-public class VoidObserverContext {
```

Comment:
Just for my own understanding why we've had `VoidObserverContext`? Looks like it doesn't do any meaningful task.

The other thing what I don't understand why we introduce another layer to pass `DeployedConfig`? Having tons of function params would justify the need but seems like it's not the case.

Reply:
This is weird indeed.

Reply:
It's not that weird to have some context implementations empty in cases where no extra contextual information is necessary. But in any case this is removed now :)